core/include/clist.h

```diff
@@ -40,23 +93,19 @@
  *
  * Used as is as reference to a list.
  *
- * clist stores a pointer to the last element of a list.  That way, both
- * appending to end of list and removing head can be done in constant time.
- *
- * Actual list objects should have a @c clist_node_t as member and then use
- * the container_of() macro in list operations.
- * See @ref thread_add_to_list() as example.
  */
 typedef list_node_t clist_node_t;
 
 /**
- * @brief inserts *new_node* into *list*
+ * @brief Appends *new_node* at the end of *list
+ *
+ * @note Complexity: O(1)
```

Comment:
if you want to use LaTeX parsing for the complexity you might as well go all in and write it `$\mathcal{O}(n)$`

see http://texblog.org/2014/06/24/big-o-and-related-notations-in-latex/ for an example